### PR TITLE
Kingfisher Image Cache

### PR DESCRIPTION
**Pull request by richardoti**
I've created a wrapper around Kingfisher to forward ASNetworkImageNode's delegate calls to it. It will improve UI as user won't see "gray squares" so frequently anymore.

### CODE REVIEW

File: Source/ImageCache.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    downloadProgressBlock: ((CGFloat) -> Void)!,
+    completion: ((CGImage!, NSError!) -> Void)!) -> AnyObject! {
+
+      let manager = KingfisherManager.sharedManager
+
+      let task = manager.retrieveImageWithURL(URL,
+        options: .None,
+        progressBlock:
+        { receivedSize, totalSize in
+          if downloadProgressBlock != nil {
```
Comment (zenangst) [inline Source/ImageCache.swift]: Could you use `guard` here?

Comment (richardoti) [inline Source/ImageCache.swift]: Does not really help here.
Guard lets me specify "bad" case immediately. But here "bad" case is just skip without calling `downloadProgressBlock `.

**Comment by RamonGilabert** [comment]
How is this in terms of performance?

**Comment by richardoti** [comment]
@RamonGilabert can't feel the difference. Without network connection it shows normal images, not grey squares.

Probably, a bit faster as now there is no need to waste time on networking. But no performance "kick".
Should help users which are on mobile, not Wi-Fi

**Pull request merged**